libs/api_common/windows_platform_common.cpp

```diff
@@ -136,10 +428,734 @@ get_attach_type_windows(const std::string& section)
+        status = open_registry_key(section_data_key, section_name, KEY_READ, &section_info_key);
+        if (status != ERROR_SUCCESS) {
+            // Registry path is not present.
+            result = EBPF_FILE_NOT_FOUND;
+            goto Exit;
+        }
+
+        program_type = (ebpf_program_type_t*)ebpf_allocate(sizeof(ebpf_program_type_t));
+        if (program_type == nullptr) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+
+        attach_type = (ebpf_attach_type_t*)ebpf_allocate(sizeof(ebpf_attach_type_t));
+        if (attach_type == nullptr) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+
+        // Read program type.
+        status = read_registry_value_binary(
+            section_info_key, EBPF_SECTION_DATA_PROGRAM_TYPE, (uint8_t*)program_type, sizeof(ebpf_program_type_t));
+        if (status != ERROR_SUCCESS) {
+            result = win32_error_code_to_ebpf_result(status);
+            goto Exit;
+        }
+
+        // Read attach type.
+        status = read_registry_value_binary(
+            section_info_key, EBPF_SECTION_DATA_ATTACH_TYPE, (uint8_t*)attach_type, sizeof(ebpf_attach_type_t));
+        if (status != ERROR_SUCCESS) {
+            result = win32_error_code_to_ebpf_result(status);
+            goto Exit;
+        }
+
+        // Read bpf program type.
+        status =
+            read_registry_value_dword(section_info_key, EBPF_PROGRAM_DATA_BPF_PROG_TYPE, (uint32_t*)&bpf_program_type);
+        if (status != ERROR_SUCCESS) {
+            bpf_program_type = BPF_PROG_TYPE_UNSPEC;
+            result = EBPF_SUCCESS;
+        }
+
+        // Read bpf attach type.
+        status =
+            read_registry_value_dword(section_info_key, EBPF_PROGRAM_DATA_BPF_PROG_TYPE, (uint32_t*)&bpf_attach_type);
+        if (status != ERROR_SUCCESS) {
+            bpf_attach_type = BPF_ATTACH_TYPE_UNSPEC;
+            result = EBPF_SUCCESS;
+        }
+
+        section_prefix = _strdup(ebpf_down_cast_from_wstring(section_name).c_str());
+        if (section_prefix == nullptr) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+
+        // We have read all the required data. Populate section definition in the global array.
+        section_definition.program_type = program_type;
+        section_definition.attach_type = attach_type;
+        section_definition.bpf_prog_type = bpf_program_type;
+        section_definition.bpf_attach_type = bpf_attach_type;
+        section_definition.section_prefix = section_prefix;
+
+        _windows_section_definitions.emplace_back(section_definition);
+    } catch (...) {
+        result = EBPF_FAILED;
+        goto Exit;
+    }
+
+Exit:
+    if (result != EBPF_SUCCESS) {
+        if (program_type) {
+            ebpf_free(program_type);
+        }
+        if (attach_type) {
+            ebpf_free(attach_type);
+        }
+        if (section_prefix) {
+            ebpf_free(section_prefix);
+        }
```

Comment:
```suggestion
        ebpf_free(program_type);
        ebpf_free(attach_type);
        ebpf_free(section_prefix);
```
improve codecov percentage